src/features/congregation/upcoming_events/upcoming_events_list/index.tsx

```diff
@@ -1,31 +1,95 @@
-import { Box } from '@mui/material';
+import { Box, Typography } from '@mui/material';
 import { UpcomingEventsListProps } from './index.types';
 import InfoTip from '@components/info_tip';
 import { IconInfo } from '@components/icons';
 import { useAppTranslation } from '@hooks/index';
-import YearlyUpcomingEvents from '../yearly_upcoming_events';
 import useUpcomingEventsList from './useUpcomingEventsList';
+import UpcomingEvent from '../upcoming_event';
 
 const UpcomingEventsList = (props: UpcomingEventsListProps) => {
   const { t } = useAppTranslation();
-  const { eventsSortedByYear } = useUpcomingEventsList(props);
+  const {
+    eventsSortedByYear,
+    stickyYearRefs,
+    stuckYearIndexes,
+    isEventExpired,
+    isAdmin,
+  } = useUpcomingEventsList(props);
 
   return (
     <Box sx={{ display: 'flex', flexDirection: 'column', gap: '16px' }}>
-      {eventsSortedByYear[0].length == 0 ? (
+      {!eventsSortedByYear.length || !eventsSortedByYear[0]?.length ? (
```

Comment:
can you explain what you mean here? not sure I get it
do you mean to use `&&` operator?
`!eventsSortedByYear[0]?.length` will throw an error if `eventsSortedByYear` is not an array
I think the first check is enough